src/modules/wallets/sagas.js

```diff
@@ -55,32 +55,53 @@ const generateWallet = function * ({ blockchainKey }) {
   }
 }
 
-// register a new wallet on blockchain
+// register wallet on blockchain
+// needed for EOS, Decent, and other blockchains that require creation of account
 const registerWallet = function * ({ blockchainKey, data }) {
+  const { registerWallet } = blockchains.get(blockchainKey)
+  try {
+    const { ok: registrationSucceeded } = yield call(registerWallet, data)
+    if (registrationSucceeded) {
+      yield put(ownA.register.succeeded({ blockchainKey }))
+    } else {
+      yield put(ownA.register.failed({ blockchainKey }))
+    }
+  } catch (error) {
+    yield put(ownA.register.errored({ error }))
+  }
+}
+
+const connectWalletWithProfile = function * () {
   try {
+    const blockchainKey = blockchains.primary.key
     const userAccessToken = yield select(S.login.userAccessToken)
     const userName = yield select(S.login.userName)
     const userID = yield select(S.login.userID)
-    const walletAddress = data.addrStr
+    const walletAddress = yield select(S.wallets.address(blockchainKey))
     const { ok: registrationSucceeded } = yield call(api.registerUser(blockchainKey), {
```

Comment:
I would like consistent wording:

either it's `registerUser` (as the name in api function used), either it's `connectWithProfile`.
It seems a bit strict, but for a bit project like this it will make huge difference in maintenability.